### DISCUSSION TITLE
Add a native support for reports and data visualization

### DISCUSSION THREAD

**Discussion by hunghvu** [payloadcms/payload#2176] [Feature Requests & Ideas]
At the moment, the Payload admin dashboard is already great at providing a minimal static view of the contents, mostly in a list-like format. However, there is a lack of support for dynamic and calculated content and data. This is very helpful for non-technical users. For example:

- The number of users (a record) registered per day (time period)
- Server resources (CPU, RAM, etc.)
- Revenue (sum of records)
- ...

It is certainly possible to DIY components, but it would be great to have native support for these reports, and data visualization.

**Comment by DanRibbens** [comment]
I agree and think it should be on our roadmap. On my backlog is looking into tools already available that Payload could integrate into our own configuration. In the end it may fall into plugins if the libraries are heavier for charts and visuals.

If anyone has suggestions, this is a good place for that conversation.

**Comment by hunghvu** [comment]
Customizable visuals are the ideal goal in my opinion, but what if we approach numeric content first? We have collections, and fields, so a built-in aggregation function can provide some calculated (numeric-only) results. [Directus has this feature, called Insights](https://docs.directus.io/app/insights.html), so it should be a good source of inspiration.

Aside from that, when looking for chart libraries, we should consider whether they support SSR or not. As far as I know, then D3 (and most D3-based) solutions do not _at default_: https://github.com/d3/d3/issues/3635.
